contracts/BinaryOption.sol

```diff
@@ -19,125 +18,50 @@ contract BinaryOption is IERC20, IBinaryOption {
-    uint internal constant _MINIMUM_BID = 1e16;
+    // Enforce a 1 cent minimum amount
+    uint internal constant _MINIMUM_AMOUNT = 1e16;
 
     /* ========== CONSTRUCTOR ========== */
 
-    constructor(address initialBidder, uint initialBid) public {
-        market = BinaryOptionMarket(msg.sender);
-        bidOf[initialBidder] = initialBid;
-        totalBids = initialBid;
-    }
-
-    /* ========== VIEWS ========== */
-
-    function _claimableBalanceOf(
-        uint _bid,
-        uint price,
-        uint exercisableDeposits
-    ) internal view returns (uint) {
-        uint owed = _bid.divideDecimal(price);
-        uint supply = _totalClaimableSupply(exercisableDeposits);
-
-        /* The last claimant might be owed slightly more or less than the actual remaining deposit
-           based on rounding errors with the price.
-           Therefore if the user's bid is the entire rest of the pot, just give them everything that's left.
-           If there is no supply, then this option lost, and we'll return 0.
-           */
-        if ((_bid == totalBids && _bid != 0) || supply == 0) {
-            return supply;
-        }
-
-        /* Note that option supply on the losing side and deposits can become decoupled,
-           but losing options are not claimable, therefore we only need to worry about
-           the situation where supply < owed on the winning side.
-
-           If somehow a user who is not the last bidder is owed more than what's available,
-           subsequent bidders will be disadvantaged. Given that the minimum bid is 10^16 wei,
-           this should never occur in reality. */
-        require(owed <= supply, "supply < claimable");
-        return owed;
-    }
-
-    function claimableBalanceOf(address account) external view returns (uint) {
-        (uint price, uint exercisableDeposits) = market.senderPriceAndExercisableDeposits();
-        return _claimableBalanceOf(bidOf[account], price, exercisableDeposits);
-    }
-
-    function _totalClaimableSupply(uint exercisableDeposits) internal view returns (uint) {
-        uint _totalSupply = totalSupply;
-        // We'll avoid throwing an exception here to avoid breaking any dapps, but this case
-        // should never occur given the minimum bid size.
-        if (exercisableDeposits <= _totalSupply) {
-            return 0;
-        }
-        return exercisableDeposits.sub(_totalSupply);
-    }
+    bool public initialized = false;
 
-    function totalClaimableSupply() external view returns (uint) {
-        (, uint exercisableDeposits) = market.senderPriceAndExercisableDeposits();
-        return _totalClaimableSupply(exercisableDeposits);
+    function initialize(
+        string memory _name,
+        string memory _symbol
+    ) public {
+        require(!initialized, "Binary Option Market already initialized");
+        initialized = true;
+        name = _name;
+        symbol = _symbol;
+        market = BinaryOptionMarket(msg.sender);
```

Comment:
Minimal proxy pattern

Reply:
All the logic related to claiming options is gone, as options are minted into users wallet on deposit